### ISSUE TITLE
No detection even with trained weights

### ISSUE THREAD

**Issue by johnny-mueller** [AlexeyAB/darknet#6503]
I have trained a custom yolov3 detector based on the MS COCO data set. For an initial test I have trained about 30000 batches and below is the corresponding graph.

Command used for training:
```python
./darknet detector -map -dont_show train cfg/coco.data cfg/yolov3_rgb.cfg -gpus 0,1 2>&1 | tee --append log.txt
```

```python
[net]
batch=64
subdivisions=16
width=416
height=416
channels=1

momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001305
burn_in=2000
max_batches = 500200
policy=steps
steps=400000,450000
scales=.1,.1
``` 

I assumed that the mAP calculations were 0.0%, because I put the trainings, validation images and annotations in the same folder. Now I am not so sure anymore about this problem with the mAP calculation (sie figure below).

I then used the gained weights for a test detection with the following command and no objects are ever detected. I also used images from the validation data set of MS COCO for testing.

```python
./darknet detector test ./cfg/coco.data ./cfg/yolov3_bw.cfg ./yolov3_rgb_30000.weights
```

As further information it is probably important to mention that I made a mistake in the cfg file (see above). I've provided the model with only one channel but as input the normal rgb images from MS COCO. During the test detection I proceeded identically: The input data were the ms COCO rgb images and the model also had 1 channel as input.

Since I don't have access to a server with multiple GPUs for the next few weeks, the question arises where exactly the error occurred or how I can test the trained weights.

Loss grapic:
![chart](https://user-images.githubusercontent.com/64581980/90380904-f3daf100-e07c-11ea-8441-cdad51d735fa.png)

Detection example:
![grafik](https://user-images.githubusercontent.com/64581980/90387055-c8a8cf80-e085-11ea-836a-efb9e718d68b.png)

Detection output:
![grafik](https://user-images.githubusercontent.com/64581980/90387101-dd856300-e085-11ea-8f86-a800987474b4.png)

Edit:
i changed for a test the flag -iou_thresh to 0.1 This did not cause any difference in detection.

**Comment by HHHcwb** [comment]
@johnny-mueller Hello, I have the same issue. Did you fixed it? Can you tell me how to solve this problem.

**Comment by stephanecharette** [comment]
@HHHcwb if you don't get an answer, also try on the Darknet/yolo discord to see if someone can help:  https://discord.gg/zSq8rtW

**Comment by HHHcwb** [comment]
@stephanecharette, Thanks a lot , I will check later.
I got my mAP information like this:
300
 detections_count = 0, unique_truth_count = 299
class_id = 0, name = smoke, ap = 0.00%         (TP = 0, FP = 0)

 for conf_thresh = 0.25, precision = -nan, recall = 0.00, F1-score = -nan
 for conf_thresh = 0.25, TP = 0, FP = 0, FN = 299, average IoU = 0.00 %

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall
 mean average precision (mAP@0.50) = 0.000000, or 0.00 %
Total Detection Time: 51 Seconds

2436
 detections_count = 0, unique_truth_count = 2604
class_id = 0, name = smoke, ap = 0.00%           (TP = 0, FP = 0)

 for conf_thresh = 0.25, precision = -nan, recall = 0.00, F1-score = -nan
 for conf_thresh = 0.25, TP = 0, FP = 0, FN = 2604, average IoU = 0.00 %

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall
 mean average precision (mAP@0.50) = 0.000000, or 0.00 %
Total Detection Time: 415 Seconds

Set -points flag:
 `-points 101` for MS COCO
 `-points 11` for PascalVOC 2007 (uncomment `difficult` in voc.data)
 `-points 0` (AUC) for ImageNet, PascalVOC 2010-2012, your custom dataset

 mean_average_precision (mAP@0.5) = 0.000000
New best mAP!

Can you tell me why the precision and F1-score is -nan?

**Comment by HHHcwb** [comment]
@stephanecharette, I want to detect the smoke and I labeled my dataset by labelme. And I covert the .xml annotation to .txt.
I use the command -show_img, I can see the pictures that I annotated. But the mAP is always zero and when I test a picture, there is no bounding box and detection.

**Comment by stephanecharette** [comment]
> I want to detect the smoke

Smoke...reminds me of #6437.

> Can you tell me why the precision and F1-score is -nan?

No, I cannot.  I'm not familiar with your project.
